inc/Engine/Preload/Frontend/FetchSitemap.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+namespace WP_Rocket\Engine\Preload\Frontend;
+
+use WP_Rocket\Engine\Preload\Controller\Queue;
+use WP_Rocket\Engine\Preload\Database\Queries\Cache;
+
+class FetchSitemap {
+
+	/**
+	 * Parse controller.
+	 *
+	 * @var SitemapParser
+	 */
+	protected $sitemap_parser;
+
+	/**
+	 * Queue instance.
+	 *
+	 * @var Queue
+	 */
+	protected $queue;
+
+	/**
+	 * DB query.
+	 *
+	 * @var Cache
+	 */
+	protected $query;
+	/**
+	 * Instantiate the class.
+	 *
+	 * @param SitemapParser $sitemap_parser Parse controller.
+	 * @param Queue         $queue Queue instance.
+	 * @param Cache         $rocket_cache DB query.
+	 */
+	public function __construct( SitemapParser $sitemap_parser, Queue $queue, Cache $rocket_cache ) {
+		$this->sitemap_parser = $sitemap_parser;
+		$this->queue          = $queue;
+		$this->query          = $rocket_cache;
+	}
+
+	/**
+	 * Parse a sitemap.
+	 *
+	 * @param string $url url from the sitemap.
+	 */
+	public function parse_sitemap( string $url ) {
+		$response = wp_safe_remote_get( $url );
+
+		if ( 200 !== wp_remote_retrieve_response_code( $response ) ) {
+			return;
+		}
+
+		$data = wp_remote_retrieve_body( $response );
+
+		$this->sitemap_parser->set_content( $data );
+		$links = $this->sitemap_parser->get_links();
```

Comment:
You might want to have a check that the body is not empty here, before passing it to the parser